src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstier/service/MandateForChange.kt

```diff
@@ -13,27 +13,29 @@ class MandateForChange(
 ) {
   fun hasNoMandate(crn: String, convictions: Collection<Conviction>): Boolean =
     convictions
-      .filter { it.sentence?.terminationDate == null }
-      .let { activeConvictions ->
-        activeConvictions.none {
-          it.sentence != null && (isCustodial(it.sentence) || hasNonRestrictiveRequirements(crn, it.convictionId))
-        }
+      .filter { currentSentence(it.sentence) }
+      .none {
+        isCustodial(it.sentence) || hasNonRestrictiveRequirements(crn, it.convictionId)
       }.also { log.debug("Has no mandate for change: $it") }
 
-  private fun isCustodial(sentence: Sentence) =
-    sentence.sentenceType.code in custodialSentences
+  private fun currentSentence(sentence: Sentence?) =
+    sentence != null && sentence.terminationDate == null
+
+  private fun isCustodial(sentence: Sentence?) =
+    sentence?.sentenceType?.code in custodialSentences
```

Comment:
nickpick but I wonder if we're being too accommodating in general, We'd know at parse time if we don't have a sentenceType for example, if we need one we can throw straight away and it will go to the DQL.

Reply:
Definitely agree with this. I did a little PR in between times to try to reduce the amount of null checking we do. It gets rid of the first check here